### PR TITLE
fix(api): misc converter changes

### PR DESCRIPTION
**Pull request by Orta21**
Ref. metriport/metriport#799

Ticket: #799

### Dependencies

- Upstream: none
- Downstream: none

### Description

Misc changes regarding MR summary and FHIR Converter

### Testing

_[Regular PRs:]_

- Local
  - [x] Convert FHIR Resources
- Production
  - [ ] Montior

### Release Plan

- [ ] Merge this

### CODE REVIEW

File: packages/api/src/external/carequality/document/shared.ts [hunk 1]
```diff
@@ -65,6 +65,7 @@ export const cqToFHIR = (
       value: docId,
     },
     docStatus,
+    status: "current",
```
Comment (Orta21) [inline packages/api/src/external/carequality/document/shared.ts:68]: This fixes the DocumentReference issue with no status

File: packages/fhir-converter/src/templates/cda/Resources/Composition.hbs [hunk 2]
```diff
@@ -81,7 +77,7 @@
                             "status":"generated",
                             "div":"<div xmlns=\"http://www.w3.org/1999/xhtml\">{{this.section.title._}}</div>",
                         }
-                    {{else}}                    
+                    {{else if this.section.code.displayName}}
```
Comment (Orta21) [inline packages/fhir-converter/src/templates/cda/Resources/Composition.hbs:80]: Cant have empty div show just dont show this if its not available

File: packages/api/src/external/carequality/document/shared.ts (hunk 1 above)
Comment (Goncharo) [reply packages/api/src/external/carequality/document/shared.ts:68]: Assuming we won't have the same issue with CW?

File: packages/core/src/external/aws/lambda-logic/bundle-to-html-adhd.ts [hunk 3]
```diff
@@ -1811,7 +1811,10 @@ function createObservationsByDate(observations: Observation[]): string {
         ${tables.observations
           .filter(observation => {
             const observationDisplay = observation.code?.coding?.find(coding => {
-              return coding.display;
+              if (coding.code !== 'UNK') {
```
Comment (Goncharo) [inline packages/core/src/external/aws/lambda-logic/bundle-to-html-adhd.ts:1814]: nit: put UNK in a const at the top of the file

File: packages/core/src/external/aws/lambda-logic/bundle-to-html-adhd.ts [hunk 4]
```diff
@@ -1833,7 +1836,10 @@ function createObservationsByDate(observations: Observation[]): string {
                 }`;
 
             const observationDisplay = observation.code?.coding?.find(coding => {
-              return coding.display;
+              if (coding.code !== 'UNK') {
```
Comment (Goncharo) [inline packages/core/src/external/aws/lambda-logic/bundle-to-html-adhd.ts:1839]: ditto

File: packages/core/src/external/aws/lambda-logic/bundle-to-html-adhd.ts [hunk 5]
```diff
@@ -1811,7 +1811,10 @@ function createObservationsByDate(observations: Observation[]): string {
         ${tables.observations
           .filter(observation => {
             const observationDisplay = observation.code?.coding?.find(coding => {
-              return coding.display;
+              if (coding.code !== 'UNK') {
+                return  coding.display;
```
Comment (Goncharo) [inline packages/core/src/external/aws/lambda-logic/bundle-to-html-adhd.ts:1815]: nit: spacing

File: packages/core/src/external/aws/lambda-logic/bundle-to-html-adhd.ts [hunk 6]
```diff
@@ -1833,7 +1836,10 @@ function createObservationsByDate(observations: Observation[]): string {
                 }`;
 
             const observationDisplay = observation.code?.coding?.find(coding => {
-              return coding.display;
+              if (coding.code !== 'UNK') {
+                return  coding.display;
```
Comment (Goncharo) [inline packages/core/src/external/aws/lambda-logic/bundle-to-html-adhd.ts:1840]: nit: spacing

File: packages/core/src/external/aws/lambda-logic/bundle-to-html-adhd.ts [hunk 7]
```diff
@@ -2512,3 +2519,11 @@ function hasClinicalRelevantData(fhirTypes: FhirTypes): boolean {
 
   return hasValues.length > 0;
 }
+
+function getValidCode(coding: Coding[] | undefined): Coding[] {
+  if (!coding) return [];
+
+  return coding.filter(coding => {
+    return coding.code && coding.code !== 'UNK'
```
Comment (Goncharo) [inline packages/core/src/external/aws/lambda-logic/bundle-to-html-adhd.ts:2527]: ditto, const

File: packages/core/src/external/aws/lambda-logic/bundle-to-html.ts [hunk 8]
```diff
@@ -1580,7 +1580,10 @@ function createObservationsByDate(observations: Observation[]): string {
         ${tables.observations
           .filter(observation => {
             const observationDisplay = observation.code?.coding?.find(coding => {
-              return coding.display;
+              if (coding.code !== 'UNK') {
```
Comment (Goncharo) [inline packages/core/src/external/aws/lambda-logic/bundle-to-html.ts:1583]: const

File: packages/core/src/external/aws/lambda-logic/bundle-to-html.ts [hunk 9]
```diff
@@ -1580,7 +1580,10 @@ function createObservationsByDate(observations: Observation[]): string {
...
+                return  coding.display;
```
Comment (Goncharo) [inline packages/core/src/external/aws/lambda-logic/bundle-to-html.ts:1584]: spacing

File: packages/core/src/external/aws/lambda-logic/bundle-to-html.ts [hunk 10]
```diff
@@ -1602,7 +1605,10 @@ function createObservationsByDate(observations: Observation[]): string {
                 }`;
 
             const observationDisplay = observation.code?.coding?.find(coding => {
-              return coding.display;
+              if (coding.code !== 'UNK') {
```
Comment (Goncharo) [inline packages/core/src/external/aws/lambda-logic/bundle-to-html.ts:1608]: const

File: packages/core/src/external/aws/lambda-logic/bundle-to-html.ts [hunk 11]
```diff
@@ -1602,7 +1605,10 @@ function createObservationsByDate(observations: Observation[]): string {
...
+                return  coding.display;
```
Comment (Goncharo) [inline packages/core/src/external/aws/lambda-logic/bundle-to-html.ts:1609]: spacing

File: packages/core/src/external/aws/lambda-logic/bundle-to-html.ts [hunk 12]
```diff
@@ -2233,3 +2240,11 @@ function getAnnualWellnessVisits(conditions: Condition[]) {
 
   return annualWellnessVisit;
 }
+
+function getValidCode(coding: Coding[] | undefined): Coding[] {
+  if (!coding) return [];
+
+  return coding.filter(coding => {
+    return coding.code && coding.code !== 'UNK'
```
Comment (Goncharo) [inline packages/core/src/external/aws/lambda-logic/bundle-to-html.ts:2248]: const

File: packages/api/src/external/carequality/document/shared.ts (hunk 1 above)
Comment (Orta21) [reply packages/api/src/external/carequality/document/shared.ts:68]: Yea we set it in CW

File: packages/fhir-converter/src/templates/cda/DataType/Period.hbs [hunk 13]
```diff
@@ -25,9 +25,15 @@
   // -------------------------------------------------------------------------------------------------
 --}}
 {
-    {{#if (or period.low period.high)}}
+    {{#if (and (and period.low period.high) (startDateLteEndDate period.low.value period.high.value) )}}
```
Comment (Goncharo) [inline packages/fhir-converter/src/templates/cda/DataType/Period.hbs:28]: are the 2 ands intentional?

File: packages/fhir-converter/src/templates/cda/DataType/Period.hbs [hunk 14]
```diff
@@ -25,9 +25,15 @@
...
         "start":"{{formatAsDateTime period.low.value}}",
         "end":"{{formatAsDateTime period.high.value}}",
+    {{else if (and (and period.low period.high) (not (startDateLteEndDate period.low.value period.high.value)) ) }}
```
Comment (Goncharo) [inline packages/fhir-converter/src/templates/cda/DataType/Period.hbs:31]: ditto 2 ands?

**Review by Goncharo — APPROVED** [review]

**Pull request merged**